bal_tools/drpc.py

```diff
@@ -0,0 +1,56 @@
+from web3 import Web3
+from bal_addresses import AddrBook
+
+DRPC_NAME_OVERRIDES = {
+    "mainnet": "ethereum",
+    "zkevm": "polygon-zkevm",
+}
+
+class Web3RpcByChain:
+    def __init__(self, DRPC_KEY):
+        self.DRPC_KEY = DRPC_KEY
+        self._w3_by_chain = {}
+
+    def __getitem__(self, chain):
+        return self._get_or_create_w3(chain)
+
+    def __getattr__(self, chain):
```

Comment:
i also added support for dot syntax, so chains can be referenced like:
```py
 Web3RpcByChain.mainnet
 # or
 Web3RpcByChain["mainnet"]
 ```
 
 since this can basically be used across the board, i think making it is worth making it more flexible